perllib/FixMyStreet/Roles/CobrandEcho.pm

```diff
@@ -142,13 +141,23 @@ sub _parse_events {
         } elsif ($type eq 'missed') {
             $self->parse_event_missed($_, $closed, $events);
         } elsif ($type eq 'bulky') {
-            my $report = $self->problems->search({ external_id => $_->{Guid} })->first;
-            my $resolved_date = construct_bin_date($_->{ResolvedDate});
-            $events->{enquiry}->{$event_type} = {
-                date => $resolved_date,
-                report => $report,
-                resolution => $_->{ResolutionCodeId},
-                state => $_->{EventStateId},
+            if (!$closed) {
+                my $report = $self->problems->search({ external_id => $_->{Guid} })->first;
+                $events->{enquiry}->{$event_type} = {
+                    report => $report,
+                    date => construct_bin_date({ DateTime => $report->get_extra_field_value('Collection_Date')}),
```

Comment:
We may fetch an event with this API call that doesn't exist in our system (if it was created outside our system), so I guess in that case we shouldn't set this object at all, as we won't know the collection date then. (Also, we could ask Brent if there is a field on a bulky collection event that is the collection date, and then use that (perhaps I should have suggested asking that in my last comment).)